src/model/document.ts

```diff
@@ -26,6 +26,14 @@ import {Import} from './import';
 import {isResolvable} from './resolvable';
 import {SourceRange} from './source-range';
 
+export class Slot {
+  name: string;
+
+  constructor(name: string) {
```

Comment:
I think `name` is optional, and represents a slot that will capture all light children that aren't captured by another slot